yast_rdp.pm

```diff
@@ -0,0 +1,90 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: this test checks that YaST2's RDP module is behaving
+#          correctly and verifying that RDP service has been
+#          successfully set.
+# Maintainer: Jun Wang <jgwang@suse.com>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use version_utils 'is_sle';
+
+my $repos = {'12-SP1' => 'http://download.suse.de/ibs/Devel:/YaST:/SLE-12-SP1/SLE-12-SP1/',
+    '12-SP1-UP' => 'http://download.suse.de/ibs/Devel:/YaST:/SLE-12-SP1/SLE-12-SP1_Update/',
+    '12-SP2'    => 'http://download.suse.de/ibs/Devel:/YaST:/SLE-12-SP2/SLE_12_SP2/',
+    '12-SP2-UP' => 'http://download.suse.de/ibs/Devel:/YaST:/SLE-12-SP2/SUSE_SLE-12-SP2_Update/',
+    '12-SP3'    => 'http://download.suse.de/ibs/Devel:/YaST:/SLE-12-SP3/SLE_12_SP3/',
+    '12-SP3-UP' => '',
+    '12-SP4'    => 'http://download.suse.de/ibs/Devel:/YaST:/SLE-12-SP4/SLE_12_SP4/',
+    '12-SP4-UP' => 'http://download.suse.de/ibs/Devel:/YaST:/SLE-12-SP4/SUSE_SLE-12-SP4_Update/'
+};
+
+
+sub check_service {
+    my ($cmd) = @_;
+
+    my $result = script_output("$cmd", proceed_on_failure => 1);
+
+    if ($result eq "active" or $result eq "enabled") {
+        return 1;
+    } else {
+        return 0;
+    }
+}
+
+sub run {
+    select_console 'root-console';
+
+    # yast2-rdp was added after SLE12SP2.
+    if (is_sle('>12-SP2') and is_sle('<15')) {
+        my $version = get_var('VERSION');
+        zypper_ar("$repos->{$version}", 'Devel-YaST-SLE');
+        if ("$version" ne '12-SP3') {
+            zypper_ar("$repos->{qq($version-UP)}", 'Devel-YaST-SLE-UP');
+        }
+    } elsif (is_sle('<12-SP3')) {
+        die "This version of product is NOT supported.";
+    }
```

Comment:
The include/exclude of test modules based on product versions is normally handled in the test schedule, better take a look in `lib/main_common.pm` and `products/opensuse/main.pm` and `products/sle/main.pm`

Reply:
Yes, I see. Thank you.